### PR TITLE
[RISCV][FMV] Remove support for negative priority

### PR DESCRIPTION
**Pull request by BeMg**
Ensure that target_version and target_clones do not accept negative numbers for the priority feature. 

Base on discussion on https://github.com/riscv-non-isa/riscv-c-api-doc/pull/85.

### CODE REVIEW

File: clang/test/SemaCXX/attr-target-version-riscv.cpp [hunk 1]
```diff
@@ -111,3 +111,9 @@ __attribute__((target_version("default"))) int invalidVerson4(void) { return 2;
 __attribute__((target_version("priority=1"))) int prioriyWithoutArch(void) { return 2; }
 // expected-error@+1 {{redefinition of 'prioriyWithoutArch'}}
 __attribute__((target_version("default"))) int prioriyWithoutArch(void) { return 2; }
+
+// expected-warning@+2 {{unsupported '-1' in the 'target_version' attribute string; 'target_version' attribute ignored}}
```
Comment (BeMg) [inline clang/test/SemaCXX/attr-target-version-riscv.cpp:115]: I plan to create another PR to make the warning messages consistent.

File: clang/lib/Sema/SemaDeclAttr.cpp [hunk 2]
```diff
@@ -3073,7 +3073,7 @@ bool Sema::checkTargetVersionAttr(SourceLocation LiteralLoc, Decl *D,
           DuplicateAttr = true;
         HasPriority = true;
         int Digit;
-        if (AttrStr.getAsInteger(0, Digit))
+        if (AttrStr.getAsInteger(0, Digit) || Digit < 0)
```
Comment (topperc) [inline clang/lib/Sema/SemaDeclAttr.cpp:3076]: getAsUnsignedInteger?

File: clang/lib/CodeGen/CodeGenFunction.cpp [hunk 3]
```diff
@@ -2919,6 +2919,8 @@ static int getPriorityFromAttrString(StringRef AttrStr) {
     }
   }
 
+  assert(Priority >= 0);
```
Comment (topperc) [inline clang/lib/CodeGen/CodeGenFunction.cpp:2922]: Use `unsigned` in this function too?

File: clang/lib/CodeGen/CodeGenFunction.cpp [hunk 4]
```diff
@@ -2909,18 +2909,16 @@ static int getPriorityFromAttrString(StringRef AttrStr) {
   AttrStr.split(Attrs, ';');
```
Comment (topperc) [inline clang/lib/CodeGen/CodeGenFunction.cpp:2909]: Should the function return `unsigned` instead of `int` now?

Comment (BeMg) [reply clang/lib/CodeGen/CodeGenFunction.cpp:2909]: Sure.

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**